agent_server.py

```
@@ -284,6 +303,45 @@ def _collect_agent_status_snapshot() -> Dict[str, Any]:
+        ).strip()
+        ts = str(
+            m.get("timestamp")
+            or m.get("time")
+            or m.get("created_at")
+            or ""
+        ).strip()
+        user_parts.append(f"{text}|{mid}|{ts}")
+    if not user_parts:
+        return None
+    payload = "\n".join(user_parts).encode("utf-8", errors="ignore")
+    return hashlib.sha1(payload).hexdigest()
```

Ruff (0.15.1)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)